File: kotlin-solutions/src/day01/Day01.kt
package day01

import readInput
import java.lang.Integer.max

fun main() {
    fun part1(input: List<String>): Int {
        var maxVal = 0
        var temp = 0
        for (num: String in input) {
            if (num != "") {
                temp += num.toInt()
            } else {
                maxVal = max(temp, maxVal)
                temp = 0
            }
        }
        return maxVal
    }

    fun part2(input: List<String>): Int {
        val calorieList = mutableListOf<Int>()
        var temp = 0
        for (num: String in input) {
            if (num != "") {
                temp += num.toInt()
            } else {
                calorieList.add(temp)
                temp = 0
            }
        }
        calorieList.sortDescending()
        return calorieList.take(3).sum()
    }

    val input = readInput("day01/Day01")
    println("Part 1 Solution: ${part1(input)}")
    println("Part 2 Solution: ${part2(input)}")
}

File: 2022/day01.py
from utils.parse_input import get_input_1d_str

data = get_input_1d_str("01")

def part1():
    temp = ans = 0
    for x in data:
        if x == "":
            ans = max(ans, temp)
            temp = 0
        else:
            temp += int(x)

    print(ans)

def part2():
    caloList = []
    temp = 0
    for x in data:
        if x == "":
            caloList.append(temp)
            temp = 0 
        else:
            temp += int(x)

    print(sum(sorted(caloList)[-3:]))

part1()
part2()
